revocation/ocsp/errors.go

```diff
@@ -0,0 +1,55 @@
+// Package OCSP provides methods for checking the OCSP revocation status of a
+// certificate chain, as well as errors related to these checks
+package ocsp
+
+import (
+	"fmt"
+	"time"
+)
+
+// RevokedError is returned when the certificate's status for OCSP is
+// ocsp.Revoked
+type RevokedError struct{}
+
+func (e RevokedError) Error() string {
+	return "certificate is revoked via OCSP"
+}
+
+// UnknownStatusError is returned when the certificate's status for OCSP is
+// ocsp.Unknown
+type UnknownStatusError struct{}
+
+func (e UnknownStatusError) Error() string {
+	return "certificate has unknown status via OCSP"
+}
+
+// OCSPCheckError is returned when there is an error during the OCSP revocation
+// check, not necessarily a revocation
+type OCSPCheckError struct {
```

Comment:
My though for that was that CheckError sounds like a function since check is a verb, whereas OCSPCheckError sounds like an error with the ocsp check, but I understand the concern. Should we consider changing this to a different name entirely? Maybe something like UnexpectedError?

Reply:
`Error`, `GenericError`, or `UnexpectedError` all sounds good to me.